packages/base-controller/src/BaseControllerV2.ts

```diff
@@ -68,7 +70,13 @@ export interface StatePropertyMetadata<T extends Json> {
 export class BaseController<
   N extends string,
   S extends Record<string, Json>,
-  messenger extends RestrictedControllerMessenger<N, any, any, string, string>,
+  messenger extends RestrictedControllerMessenger<
+    N,
```

Comment:
If we're doing a refactor, I would prefer moving away from non-descriptive single-letter generics. I have no idea what `N` and `S` are without looking at more of the source code currently.

Reply:
Good call! I got rid of all the remaining single-letter generics here: https://github.com/MetaMask/core/pull/1890/commits/08413a9ed9b16d92e4c782281f99b45aa4ef71c9

Reply:
Extracted to new PR: https://github.com/MetaMask/core/pull/2030/